### PR TITLE
fix(connector): [BOA, Cybersource] capture error_code

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Resolves [#3739](https://github.com/juspay/hyperswitch-cloud/issues/3739)

## How did you test it?

1.Create a failed payment using Cybersource, ensuring that the error_code includes the specific error reason (connector response must include error.reason)

Screenshot 2024-01-04 at 1 48 12 PM

3.Create a failed payment with BOA, ensuring that the error_code includes the specific error reason (connector response must include error.reason)

Screenshot 2024-01-04 at 1 47 44 PM

## Test Case
Failed card number for both connectors: `412345678912345678914`
1.Create a failed payment using Cybersource, ensuring that the error_code includes the specific error reason (connector response must include error.reason)
2.Create a failed refund using Cybersource, ensuring that the error_code includes the specific error reason (connector response must include error.reason)
3.Create a failed payment using BOA, ensuring that the error_code includes the specific error reason (connector response must include error.reason)
4.Create a failed refund using BOA, ensuring that the error_code includes the specific error reason (connector response must include error.reason)

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/bankofamerica/transformers.rs [hunk 1]
```diff
@@ -1262,3 +1219,28 @@ pub struct ErrorInformation {
 pub struct AuthenticationErrorInformation {
     pub rmsg: String,
 }
+
+impl From<(&Option<BankOfAmericaErrorInformation>, u16, String)> for types::ErrorResponse {
+    fn from(value: (&Option<BankOfAmericaErrorInformation>, u16, String)) -> Self {
```
Comment (ArjunKarthik) [inline crates/router/src/connector/bankofamerica/transformers.rs:1224]:
```suggestion
    fn from((error_data,status_code, transaction_id): (&Option<BankOfAmericaErrorInformation>, u16, String)) -> Self {
```

File: crates/router/src/connector/cybersource/transformers.rs [hunk 2]
```diff
@@ -1787,3 +1744,28 @@ pub struct ErrorInformation {
 pub struct AuthenticationErrorInformation {
     pub rmsg: String,
 }
+
+impl From<(&Option<CybersourceErrorInformation>, u16, String)> for types::ErrorResponse {
+    fn from(value: (&Option<CybersourceErrorInformation>, u16, String)) -> Self {
```
Comment (ArjunKarthik) [inline crates/router/src/connector/cybersource/transformers.rs:1749]:
```suggestion
    fn from((error_data,status_code, transaction_id): (&Option<CybersourceErrorInformation>, u16, String)) -> Self {
```

**Review by ArjunKarthik — APPROVED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

File: crates/router/src/connector/bankofamerica/transformers.rs [hunk 3]
```diff
@@ -798,12 +755,16 @@ impl<F>
             }
             BankOfAmericaPaymentsResponse::ErrorInformation(error_response) => Ok(Self {
                 response: Err(types::ErrorResponse {
-                    code: consts::NO_ERROR_CODE.to_string(),
+                    code: error_response
```
Comment (srujanchikke) [inline crates/router/src/connector/bankofamerica/transformers.rs:758]: Is this value dynamic or static ?

Comment (AkshayaFoiger) [reply crates/router/src/connector/bankofamerica/transformers.rs:758]: This value is dynamic

Comment (srujanchikke) [reply crates/router/src/connector/bankofamerica/transformers.rs:758]: Error message should be unique static enums so please map it to error reason according to [cybersource doc](https://support.cybersource.com/knowledgebase/knowledgearticle/?code=KA-04103)

**Review by srujanchikke — APPROVED** [review]

**Review by srujanchikke — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by deepanshu-iiitu — APPROVED** [review]

**Pull request merged**